bundles/org.openhab.binding.sunsynk/src/main/java/org/openhab/binding/sunsynk/internal/api/DeviceController.java

```diff
@@ -0,0 +1,205 @@
+        this.batterySettings.buildLists();
+    }
+
+    private void getGridRealTime() throws IOException, JsonSyntaxException {
+        String response = apiGetMethod(makeURL("api/v1/inverter/grid/" + this.sn + "/realtime?sn=") + this.sn,
+                APIdata.static_access_token);
+        Gson gson = new Gson();
+        this.grid = gson.fromJson(response, Grid.class);
+        this.grid.sumVIP();
+    }
+
+    private void getBatteryRealTime() throws IOException, JsonSyntaxException {
+        String response = apiGetMethod(
+                makeURL("api/v1/inverter/battery/" + this.sn + "/realtime?sn=" + this.sn + "&lan"),
+                APIdata.static_access_token);
+        Gson gson = new Gson();
+        this.realTimeBattery = gson.fromJson(response, Battery.class);
+    }
+
+    private void getInverterACDCTemperatures() throws IOException, JsonSyntaxException {
+        String date = getAPIFormatDate();
+        String response = apiGetMethod(
+                makeURL("api/v1/inverter/" + this.sn + "/output/day?lan=en&date=" + date + "&column=dc_temp,igbt_temp"),
+                APIdata.static_access_token);
+        Gson gson = new Gson();
+        this.inverter_day_temperatures = gson.fromJson(response, Daytemps.class);
+        this.inverter_day_temperatures.getLastValue();
+    }
+
+    private void getRealTimeIn() throws IOException, JsonSyntaxException { // Get URL Respnse
+        String response = apiGetMethod(makeURL("api/v1/inverter/" + this.sn + "/realtime/input"),
+                APIdata.static_access_token);
+        Gson gson = new Gson();
+        this.realTimeDataIn = gson.fromJson(response, RealTimeInData.class);
+        this.realTimeDataIn.sumPVIV();
+    }
+
+    public void sendSettings(@Nullable Settings settings) throws SunSynkSendCommandException {
+        String body = settings.buildBody();
+        sendCommandToSunSynk(body);
+    }
+
+    public void sendCommandToSunSynk(String body) throws SunSynkSendCommandException {
+        String path = "api/v1/common/setting/" + this.sn + "/set";
+
+        try {
+            apiPostMethod(makeURL(path), body, APIdata.static_access_token);
+        } catch (IOException e) {
+            logger.debug("Failed to send to Inverter API: {} ", e.getMessage());
+            int found = e.getMessage().indexOf("Authentication challenge without WWW-Authenticate header");
+            if (found > -1) {
+                throw new SunSynkSendCommandException("Authentication token failed", e);
+            }
+            throw new SunSynkSendCommandException("Unknown athentication fail", e);
+        }
+    }
+
+    private String apiPostMethod(String httpsURL, String body, String access_token) throws IOException {
+        Properties headers = new Properties();
+        headers.setProperty("Accept", "application/json");
+        headers.setProperty("Authorization", "Bearer " + access_token);
+        headers.setProperty("Requester", "www.openhab.org"); // optional
```

Comment:
If this is optional, i would prefer to remove it as there is no perpose. and i guess the local URI is a better fit.

Reply:
okay, also done in AccountController.java